lib/manager.rb

```diff
@@ -0,0 +1,228 @@
+require 'awesome_print'
+module Hotel
+  class Manager
+    attr_reader :all_rooms, :all_reservations, :nigth_rate, :blocks
+
+    def initialize
+      @all_rooms = initialize_all_rooms
+      @all_reservations = Array.new()
+      @nigth_rate = 200
+      @blocks = Array.new()
+    end
+
+    #______________Reports_for_management:
+
+    def available_rooms(date1, date2)
+      # Validate inputs:
+      validate_date_input(date1, date2)
+
+      # Return only available rooms:
+      available_rooms = all_rooms
+      date_requested_range = get_date_range(date1, date2)
+
+      date_requested_range.each do |date|
+        available_rooms.delete_if {|room| room.ocupied_on.include?(date) }
+      end
+      return available_rooms
+    end
+
+    def list_reservations_at(reuqested_date)
+      validate_date_input(reuqested_date)
+
+      date_reservations = []
+
+      all_reservations.each do |reservation|
+        start_date = reservation.start_date
+        end_date = reservation.end_date
+
+        reservation_date_range = (start_date..end_date).map{ |date| date}
+
+        date_reservations << reservation if reservation_date_range.include?(Date.parse(reuqested_date))
+      end
+      return date_reservations
+    end
+
+    def total_cost_of_reservation(reservation_id)
+      this_reserv = all_reservations.find {|reserv| reserv.id == reservation_id}
+      this_reserv.nil? ? ArgumentError : this_reserv.cost
```

Comment:
Instead of 
```ruby
  this_reserv.nil? ? ArgumentError : this_reserv.cost
```

It should be

```ruby
this_reserv.nil? ? ArgumentError.new "No reservation at this date!" : this_reserv.cost
```